client/client.go

```diff
@@ -80,13 +79,12 @@ type OpAMPClient interface {
 	// Once stopped OpAMPClient cannot be started again.
 	Stop(ctx context.Context) error
 
-	// SetAgentAttributes sets attributes of the Agent. The attributes will be included
+	// SetAgentDescription sets attributes of the Agent. The attributes will be included
 	// in the next status report sent to the server.
-	// MAY be called before Start(), in which case the attributes will be
-	// included in the first start report after  connection is established.
-	// MAY also be called after Start(), in which case the attributes will be included
+	// MUST NOT be called before Start().
```

Comment:
What's the reason for this?

Reply:
The initial value is provided in the `settings` passed to Start(). However, you raise a good question, perhaps we remove it from there and mandate that SetAgentDescription() is called before Start(). This way we will have only one way to set it.
Let me think about it.

Reply:
I will address this in a separate PR.